### ISSUE TITLE
Too Many Request

### ISSUE THREAD

**Issue by iCueto** [sirupsen/airrecord#27]
The library return 429 code status when I use relations.

How minify or handle that problem when I use the librery as a Model?

**Comment by chrisfrank** [comment]
@sirupsen do you think you'd accept a PR to fix this issue? I'm happy to work on it.

I think the problem is that `table.[]` calls `find` on the associated table, making a new HTTP request for each associated record. I'd expect to hit Airtable's rate limit any time a record has more than five associations.

I think we could fetch all associated records in one request by doing something like this:

```ruby
def find_many(ids = [])
  # prepare an array of Airtable RECORD_ID() functions
  id_formulas = ids.map { |id| "RECORD_ID() = '#{id}'"}
  # join that array into an Airtable OR formula
  formula = "OR(#{id_formulas.join(',')})"
  # fetch the records from Airtable in a single HTTP request
  records = self.class.all(filter: formula)
  # return them in reverse order, b/c the Airtable API returns them backwards??
  records.sort_by { |record| -ids.index(record.id) }
end
```

**Comment by chrisfrank** [comment]
Great, on it!

I'd propose a slightly different association API while I'm at it, but it would be a breaking change, so I'm totally happy to work within the current API if you prefer.

Right now table.[] is a fairly complex method, in that it behaves differently for associations than for regular fields. Before I make it more complex with this patch, I'd love to refactor it such that:

```ruby
class Tea < Airrecord::Table
  has_many :brews, class: 'Brew', column: 'Brews'
end

@tea = Tea.find('recABC')

# calling record[association] would return the raw Airtable field, an array of IDs:
@tea['Brews'] #=> ["rec123", "rec345"]

# calling record.association would return an array of associated records:
@tea.brews => [<Brew @id: 'rec123'>, <Brew @id: 'rec345'>]
```

This API would be easier to work with under the hood, and to the end user it would more closely mimic the ActiveRecord associations API. But again, I'm happy to work within the current API if that seems best to you.

FWIW, the implementation would look like this:

```ruby
def has_many(method_name, config)
  define_method(method_name.to_sym) do
    ids = self[config.fetch(:column)] || []
    table = Kernel.const_get(config.fetch(:class))
    config[:single] ? table.find(ids.first) : table.find_many(ids)
  end
end
```

**Comment by sirupsen** [comment]
That makes a ton of sense to me, I'm ok with breaking backwards compatibility. < 1.0 is all about going from "oh this API makes sense to me" in isolation to others using it and going to "if we make these changes it makes more sense." 👍

**Comment by sirupsen** [comment]
With these types of changes we can likely lock in a 1.0

**Comment by chrisfrank** [comment]
@iCueto this fix has been merged, and should be available on rubygems as part of the 1.0 release.
